pkg/controller/utils/pod/pod.go

```diff
@@ -101,12 +101,13 @@ func GetPodConditionFromList(conditions []v1.PodCondition, conditionType v1.PodC
 
 // IsPodRestarting checks if a pod in the Canary deployment is restarting
 // This returns the "reason" for the pod with the most restarts
-func IsPodRestarting(pod *v1.Pod) (bool, datadoghqv1alpha1.ExtendedDaemonSetStatusReason) {
-	var maxRestartCount int
+func IsPodRestarting(pod *v1.Pod, maxRestartCount int) (bool, datadoghqv1alpha1.ExtendedDaemonSetStatusReason) {
```

Comment:
I know this function wasn't introduced in this PR but it would be great to unit test it